code/modules/projectiles/projectile.dm

```diff
@@ -41,46 +50,26 @@
 	var/forcedodge = 0
 	// 1 to pass solid objects, 2 to pass solid turfs (results in bugs, bugs and tons of bugs)
 
-/obj/item/projectile/New()
-	permutated = list()
-	return ..()
+	var/proj_hit = 0
 
 /obj/item/projectile/proc/Range()
-	range--
-	if(range <= 0 && loc)
-		on_range()
+	if(range)
+		range--
+		if(range <= 0 && loc)
+			on_range()

```

Comment:
@Iamgoofball Is it me or did you just copy pasted paprika's code without taking into account the PRs that were merged since then?